### ISSUE TITLE
[a11y]: File uploader item icon is tabbable & has incorrect role.

### ISSUE THREAD

**Issue by Akshat55** [carbon-design-system/carbon#12101]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

Windows, MacOS

### Package version

1.12.0

### React version

17/18

### Automated testing tool and ruleset

IBM equal access accessibility checker - Latest Deployment

### Description

The tabbable element's role 'img, graphics-document' is not a widget role. The checkmark--filled icon is tabbable (`tabindex="0"`) and the role assigned to the svg is img (`role="img"`).

This accessibility issue can be observed via [storybook](https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader-item&args=status:complete). Set status to `complete` & open the canvas in a new [tab (With only File uploader item story)](https://react.carbondesignsystem.com/iframe.html?args=status:complete&id=components-fileuploader--file-uploader-item&viewMode=story) & run IBM equal accessibility checker.

[Permalink to icon in code.](https://github.com/carbon-design-system/carbon/blob/70eefeee38cf2516ea742d7e70b82e8b4ffde4f4/packages/react/src/components/FileUploader/Filename.js#L36)

![image](https://user-images.githubusercontent.com/38994122/190449941-b9e1c80a-2761-4869-8d03-8797b003cd7f.png)

Have ran the checker on firefox & chrome only. 

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=status:complete&id=components-fileuploader--file-uploader-item&viewMode=story

### Steps to reproduce

* Set File uploader item status to `complete`
* Run checker

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Akshat55** [comment]
I'm not even sure if the icon should be tabbable 🤔

**Comment by tw15egan** [comment]
It should probably only be tabbable if the status is set to `edit`, since that is the `x` button to delete the file
